src/coreclr/nativeaot/Runtime/unix/unixasmmacrosarm64.inc

```diff
@@ -3,6 +3,13 @@
 
 #include "AsmOffsets.inc"
 
+// Apple Clang 16 incorrectly check constraints for .alt_entry and .cfi_adjust_cfa_offset.
+// This was reported and fixed upstream: https://github.com/llvm/llvm-project/issues/97116
+// Until a fixed version of Xcode Clang is available we avoid emiting epilog CFI codes.
+#if __clang_major__ >= 16 && __clang_major__ < 17 && defined(__apple_build_version__)
+#define NO_CFI_EPILOG 1
```

Comment:
nit: `#define NO_CFI_EPILOG` (drop 1, since we are only checking defined/!defined)